include/thread_pool/thread_pool.hpp

```diff
@@ -150,17 +186,17 @@ namespace tp
     {
         const auto ok = try_post(std::forward<Handler>(handler));
         assert(ok);
+        _tp_unused(ok);
     }
 
     template <typename TSettings>
-    inline Worker& ThreadPool<TSettings>::getWorker()
+    inline typename ThreadPool<TSettings>::FixedWorker& ThreadPool<TSettings>::getWorker()
```

Comment:
I realized this is a pure C++14 feature, so I will revert this and move the inline method inside the of the class to avoid the syntax issue:

```
       FixedWorker & getWorker()
        {
            auto id = FixedWorker::getWorkerIdForCurrentThread();

            if(id > m_workers.size())
            {
                id = m_next_worker.fetch_add(1, std::memory_order_relaxed) % m_workers.size();
            }

            return *m_workers[id];
        }
```